### ISSUE TITLE
Unable to access outputs of other templates

### ISSUE THREAD

**Issue by VarunkumarManohar** [GoogleCloudPlatform/deploymentmanager-samples#103]
I have a deployment manager script as follows:
cluster.py creates a kubernetes cluster and when the script was run only for the k8 cluster creation, it was successful -- so it means the cluster.py had no issues in creation of a k8 cluster

cluster.py also exposes ouputs:
A small snippet of the cluster.py is as follows:
```
outputs.append({
        	'name': 'v1endpoint' ,
        	'value': type_name + type_suffix })
    
return {'resources': resources, 'outputs': outputs}
```
    
If I try to access the exposed output inside dmnginxservice resource below as $(ref.dmcluster.v1endpoint) I get an error as resource not found
```
imports:
- path: cluster.py 
- path: nodeport.py

resources:
- name: dmcluster
  type: cluster.py
  properties:
   zone: us-central1-a
      
- name: dmnginxservice
  type: nodeport.py
  properties:
   cluster: $(ref.dmcluster.v1endpoint)
   image: gcr.io/pr1/nginx:latest
   port: 342
   nodeport: 32123
    
```

```
ERROR: (gcloud.deployment-manager.deployments.create) Error in Operation [operation-1519960432614-566655da89a70-a2f917ad-69eab05a]: errors:
- code: CONDITION_NOT_MET
  message: Referenced resource yaml%dmcluster could not be found. At resource
    gke-cluster-dmnginxservice.

```

**Comment by krwenholz** [comment]
Hi @VarunkumarManohar,

I'm having trouble reproducing this. Do you have a minimal reproduction? Also, have you created a .schema file for cluster.py specifying the outputs?

Thanks,

Kyle

**Comment by krwenholz** [comment]
I'm guessing you've already found the docs [here](https://cloud.google.com/deployment-manager/docs/configuration/expose-information-outputs#using_outputs_from_other_templates), but [here](https://github.com/GoogleCloudPlatform/deploymentmanager-samples/blob/e01e6b634f8979fd9d500cec4e2401cc552e1c07/examples/v2/waiter/config.yaml) is an example where a template output is used in the same deployment. Hopefully that will help.

**Comment by VarunkumarManohar** [comment]
Hi @krwenholz, here is a detailed description of the issue I am facing:
[https://issuetracker.google.com/issues/74095388](https://issuetracker.google.com/issues/74095388)


The expanded config seems to have problem referencing the resources from the previous step.

**Comment by VarunkumarManohar** [comment]
@krwenholz  I would like to know if you were able to reproduce the issue I am facing. If not, please let me know and I will update the gists appropriately. Appreciate your help.

**Comment by krwenholz** [comment]
Hi @VarunkumarManohar,

I'm working on reproducing using your gists. I'll try to get back to you by the end of the day.

At first glance, it doesn't look like your template is making any obvious mistakes.

Thanks,

Kyle

**Comment by krwenholz** [comment]
I managed to reproduce your issue on a smaller config. I suspect there's an undocumented limitation with template outputs. I'll be able to ask a few coworkers with more knowledge tomorrow and hopefully have a suggestion after that.

**Comment by VarunkumarManohar** [comment]
@krwenholz  thanks for your update.

**Comment by krwenholz** [comment]
Hi @VarunkumarManohar,

This is looking more and more like a bug on our end. To unblock you, I managed to get a line like 

    cluster_type = ''.join([context.env['project'] + '/', '$(ref.dmcluster.v1endpoint'])

working. It requires you hardcode the reference in your template, but it should work for now (and maintains some level of decoupling). Or you could change the output to include the project, so something like

    'value': context.env['project'] + '/' + type_name + type_suffix

It's the combination of project and cluster "references" together that causes the error you see. The `yaml%` injection is normal behavior in terms of indicating a reference couldn't be found (albeit confusing normal behavior). I'll keep this issue open until we close our internal bug for the paired references.

Sorry for your trouble on this one.

**Comment by VarunkumarManohar** [comment]
@krwenholz  I could not resolve the issue based on your inputs.
Could you please share a gist if possible ? Thanks, Varun

**Comment by VarunkumarManohar** [comment]
@krwenholz  I notice another issue when attempting to create a nodeport service soon after the creation of a cluster in the same config. The error is basically about type not found. 


Ideally when $ref. is used inside the nodeport service creation, the resource creation has to wait until the dependent cluster is up and running. It seems like this wait on the cluster to be up is not currently is not happening


However when the cluster creation and nodeport service creation are run in isolation everything runs fine.

**Comment by VarunkumarManohar** [comment]
@krwenholz I have created another related issue:
#110

**Comment by krwenholz** [comment]
Hi @VarunkumarManohar,

Sorry there was a typo in that first suggestion. The example I've used for testing requires a few internal types, so I won't be able to share a gist right now. (If that becomes absolutely necessary though we can work that out.) To be clear, I think you have two options for fixing this.

1. You can hard-code the ref in `nodeport.py` and set cluster_type with
    `cluster_type = ''.join([context.env['project'], '/', '$(ref.dmcluster.v1endpoint)'])`

2. You can leave `nodeport.py` the same and change the definition of `v1endpoint` in `cluster.py` to include the project like this (or some variant)
    `'value': context.env['project'] + '/' + type_name + type_suffix`

If neither of those work for you, let me know what it looks like and what the error message is and we can go from there.

Regarding the waiting issue, I think that's a known problem in Google's K8s, but I'll let the current on-call address it in your separate issue.

**Comment by VarunkumarManohar** [comment]
@krwenholz  thanks you
